libvast/vast/table_slice.hpp

```diff
@@ -143,6 +143,13 @@ class table_slice final {
   /// @pre `row < rows() && column < columns()`
   data_view at(size_type row, size_type column) const;
 
+  /// Retrieves data by specifying 2D-coordinates via row and column.
+  /// @param row The row offset.
+  /// @param column The column offset.
+  /// @param t The type of the value to be retrieved.
+  /// @pre `row < rows() && column < columns()`
```

Comment:
The precondition needs to be adapted to include the assertion that checks that the given type is correct.

Additionally, can you expand on the general description of this function to include the motivation? E.g., to be used when the type is already known for faster access.